### ISSUE TITLE
getinput fails for multiple choice problem accepting multiple answers

### ISSUE THREAD

**Issue by mabynke** [UCL-INGI/INGInious#334]
I am writing a run script to grade a multiple choice problem that accepts multiple answers. However, I am unable to get the student's input, because `getinput problemid` gives me the following:

```
Traceback (most recent call last):
  File "/usr/bin/getinput", line 26, in <module>
    sys.stdout.buffer.write(result)
TypeError: a bytes-like object is required, not 'list'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/getinput", line 28, in <module>
    sys.stdout.buffer.write(result.encode("utf-8"))
AttributeError: 'list' object has no attribute 'encode'
```

`/usr/bin` seems to be failing because it does not expect `result` being a list – but according to the [documentation](https://inginious.readthedocs.io/en/v0.5/teacher_doc/task_file.html#multiple-choice-problems) it should indeed be a list when accepting multiple answers, so this seems like a bug. Or maybe there is another way I should rather be getting this input?

For the record, it works fine when the problem accepts only one answer (as `result` is just an integer in that case).

**Comment by GuillaumeDerval** [comment]
I think you can still access it via the Python API. 

```
from inginious import input
answer = input.get_input("problemid")
```

**Comment by anthonygego** [comment]
It seems that this case was never managed actually. The current workaround is to use the Python API as @GuillaumeDerval suggested. 

We'll need to decide whether to return Bash syntax for lists and dictionaries instead, which would limit compatibility with other shells, or to manage this case in the Python API itself and add some arguments to the shell API...

**Comment by GuillaumeDerval** [comment]
For now, getinput() in the python API indicates that it always returns a str or a byte object in Py3.
https://github.com/UCL-INGI/INGInious/blob/master/base-containers/base/inginious/input.py#L35

In reality, it returns either a str, a byte obj, or a list. It could even return other things as it is mostly defined by problem types. 

We should clearly indicates everything there. My proposition is
* to indicate that the Python API could return anything, and to check the problem type documentation for more info
* to ensure that getinput (bash API) always returns a string. Particularly, we can simply call the `str()` function on all objects that are not str/byte objs. This means that the returned syntax for list would be something like `[1, 2, 3]`.

**Comment by micahstubbs** [comment]
Have encountered what appears to be this issue as a student, where the course admin is unable to make the correct answer for a free text input field case-insensitive, ie accepting `answer` and `Answer` and `ANSWER` all as valid correct answers.

**Comment by anthonygego** [comment]
Fixed by a534cb7
